### ISSUE TITLE
:count in karras.collections/fetch is broken.

### ISSUE THREAD

**Issue by bhenry** [wilkes/karras#4]
it appears that fetch doesn't work with the count flag.

https://github.com/wilkes/karras/blob/0ed5292e64f2afd97dff754711f428d8dd49d733/src/karras/collection.clj#L144

i tried this:

```
(fetch Client nil :count true) ;;it was inside with-mongo-request
```

i got back:

```
java.lang.IllegalArgumentException: Don't know how to create ISeq from: java.lang.Integer
[Thrown class java.lang.RuntimeException]
```

this however works fine:

```
(fetch Client nil :limit 1)
```

NOTE: I ended up using karras.entity/count-instances, but didn't know if the above was expected to work as it is still in the code.

**Comment by wilkes** [comment]
that's weird since count-instances ultimately uses 
`(karras.collection/fetch coll criteria :count true)`

Maybe karras.entity/fetch is broken.  I'll look into it.

**Comment by bhenry** [comment]
you're right. karras.entity/fetch is trying to map (partial make type) over the count.

**Comment by wilkes** [comment]
Ah, I need to beef up the tests.  Use count-instances in the mean time.
